x-pack/plugins/security_solution/public/resolver/store/data/selectors.ts

```diff
@@ -665,3 +665,74 @@ export const panelViewAndParameters = createSelector(
 export const nodeEventsInCategory = (state: DataState) => {
   return state.nodeEventsInCategory?.events ?? [];
 };
+
+export const userCanRequestMoreNodeEventsInCategory = createSelector(
+  (state: DataState) => state.nodeEventsInCategory,
+  panelViewAndParameters,
+  /* eslint-disable-next-line no-shadow */
+  function (nodeEventsInCategory, panelViewAndParameters) {
+    if (
+      nodeEventsInCategory !== undefined &&
+      nodeEventsInCategoryModel.isRelevantToPanelViewAndParameters(
+        nodeEventsInCategory,
+        panelViewAndParameters
+      )
+    ) {
+      return nodeEventsInCategory.cursor !== null;
```

Comment:
I believe this should return false if `lastCursorRequested === nodeEventsInCategory.cursor` because in that case, a request is already in progress. You might be able to re-use `isLoadingMoreNodeEventsInCategory` here.

Reply:
Yes, that is true, but the purpose of this selector is just to show/hide the button. When the button is visible and the request in progress, the EuiButton loading state disables any click events.

Reply:
I'm not following you.

Reply:
I think it was just a misunderstanding? You should be able to just use `isLoadingMoreNodeEventsInCategory`

Reply:
I now understand the logic. You want to show a button when the user can 'load more' **or** when the app is loading more. However I don't think the name of this selector matches its behavior. 

If you'd rather not change your implementation, can you change the name of this selector? 

Otherwise could you change the return to:
```
return nodeEventsInCategory.cursor !== null && nodeEventsInCategory.cursor !== lastCursorRequested
```
That would make the name of the selector match its return.

And then show the "Load More" / "Loading" button when:
```
userCanRequestMoreNodeEventsInCategory || isLoadingMoreNodeEventsInCategory
```

I think that design will be simpler to understand and modify in the future.